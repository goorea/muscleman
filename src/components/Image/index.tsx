import { isEqual } from 'lodash';
import React from 'react';
import FastImage, { ImageStyle, Source } from 'react-native-fast-image';

type P = {
  source: Source | number;
  style?: ImageStyle;
};

const Image: React.FC<P> = ({ source, style }) => {
  return (
    <FastImage
      source={source}
      style={style}
      resizeMode={FastImage.resizeMode.cover}
    />
  );
};

export default React.memo(Image, isEqual);
