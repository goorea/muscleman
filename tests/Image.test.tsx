import { render } from '@testing-library/react-native';
import React from 'react';

import Image from '@src/components/Image';

describe('Image 컴포넌트', () => {
  it('렌더링이 올바르게 된다', () => {
    const { toJSON } = render(
      <Image source={{ uri: 'https://via.placeholder.com/40x40' }} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
