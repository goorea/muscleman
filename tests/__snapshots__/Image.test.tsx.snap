// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image 컴포넌트 렌더링이 올바르게 된다 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <FastImageView
    resizeMode="cover"
    source={
      Object {
        "uri": "https://via.placeholder.com/40x40",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
